src/shared/components/query/QueryStore.ts

```diff
@@ -1004,14 +1007,6 @@ export class QueryStore {
             );
         },
         default: [],
-        onResult: () => {
```

Comment:
@BasLee this is the offending onResult.  It depends upon this._selectedSampleListId, so when it mutates that value it re-fires invoke and is circular.

Reply:
I believe that it was necessary in order to clear the selection when user selected a different study.  This can/should be handled on read by checking to see if the _selectedSampleListId matches selectedStudies

Reply:
the weird thing was, that the infinite loop even existed without any code in the `onResult`: it seemed like the presence of onResult itself was causing the loop

Reply:
the weird thing was, that the infinite loop even existed without any code in the onResult: it seemed like the presence of onResult itself was causing the loop

Reply:
interesting.  i will test that.

Reply:
i see that you are right.  in my testing, the problem is that mobxpromise's onResult references this.result in order to pass it to the callback:

![image](https://github.com/cBioPortal/cbioportal-frontend/assets/186521/102bbb21-b230-4f2e-8c3a-44837cc3130a)
 
when i use result instead of this.result, the problem goes away.   will have to make this change to the library.  

nevertheless, i think getting rid of onResult in general is a good idea as it is almost never actually necessary.